In [1]:
import pandas as pd

# Import 2021 Enchantments Lottery Data
raw_df = pd.read_csv("./2021_results.csv", header=1, parse_dates=True)

# Take a quick look at the data
raw_df.head()

Preferred Entry Date 1   Preferred Division 1  \
0              7/25/2021  Core Enchantment Zone   
1              8/12/2021  Core Enchantment Zone   
2              7/30/2021  Core Enchantment Zone   
3               7/1/2021  Core Enchantment Zone   
4              6/21/2021          Colchuck Zone   

   Minimum Acceptable Group Size 1  Maximum Requested Group Size 1  \
0                                7                               7   
1                                8                               8   
2                                4                               4   
3                                4                               4   
4                                2                               2   

  Preferred Entry Date 2   Preferred Division 2  \
0               8/1/2021  Core Enchantment Zone   
1              8/12/2021          Colchuck Zone   
2              8/14/2021  Core Enchantment Zone   
3               6/9/2021              Snow Zone   
4              6/28/2021          Colchuck Zone   

   Minimum Acceptable Group Size 2  Maximum Requested Group Size 2  \
0                              7.0                             7.0   
1                              8.0                             8.0   
2                              4.0                             4.0   
3                              4.0                             4.0   
4                              2.0                             2.0   

  Preferred Entry Date 3     Preferred Division 3  \
0              7/25/2021                Snow Zone   
1              8/12/2021  Eightmile/Caroline Zone   
2              7/16/2021    Core Enchantment Zone   
3               7/7/2021            Colchuck Zone   
4              7/13/2021             Stuart  Zone   

   Minimum Acceptable Group Size 3  Maximum Requested Group Size 3  \
0                              8.0                             8.0   
1                              8.0                             8.0   
2                              7.0                             7.0   
3                              4.0                             4.0   
4                              2.0                             2.0   

  Results Status  Awarded Preference Awarded Entry Date  \
0   Unsuccessful                 NaN                NaN   
1   Unsuccessful                 NaN                NaN   
2   Unsuccessful                 NaN                NaN   
3   Unsuccessful                 NaN                NaN   
4   Unsuccessful                 NaN                NaN   

  Awarded Entrance Code/Name  Awarded Group Size  
0                        NaN                 NaN  
1                        NaN                 NaN  
2                        NaN                 NaN  
3                        NaN                 NaN  
4                        NaN                 NaN

In [2]:
# View descriptive statistics
raw_df.describe()

Minimum Acceptable Group Size 1  Maximum Requested Group Size 1  \
count                     36695.000000                    36695.000000   
mean                          4.990026                        4.990026   
std                           2.102917                        2.102917   
min                           1.000000                        1.000000   
25%                           4.000000                        4.000000   
50%                           4.000000                        4.000000   
75%                           6.000000                        6.000000   
max                           8.000000                        8.000000   

       Minimum Acceptable Group Size 2  Maximum Requested Group Size 2  \
count                     36273.000000                    36273.000000   
mean                          4.982384                        4.982384   
std                           2.103137                        2.103137   
min                           1.000000                        1.000000   
25%                           4.000000                        4.000000   
50%                           4.000000                        4.000000   
75%                           6.000000                        6.000000   
max                           8.000000                        8.000000   

       Minimum Acceptable Group Size 3  Maximum Requested Group Size 3  \
count                     35674.000000                    35674.000000   
mean                          4.982817                        4.982817   
std                           2.109859                        2.109859   
min                           1.000000                        1.000000   
25%                           4.000000                        4.000000   
50%                           4.000000                        4.000000   
75%                           6.000000                        6.000000   
max                           8.000000                        8.000000   

       Awarded Preference  Awarded Group Size  
count         2445.000000         2445.000000  
mean             1.761145            4.654397  
std              0.823278            2.141760  
min              1.000000            1.000000  
25%              1.000000            3.000000  
50%              2.000000            4.000000  
75%              2.000000            6.000000  
max              3.000000            8.000000

In [3]:
# View the minimum and maximum dates for each entry
raw_df["Preferred Entry Date 1"] = pd.to_datetime(raw_df["Preferred Entry Date 1"])

print(
    "Preferred Entry Date 1:",
    raw_df["Preferred Entry Date 1"].min(),
    "to",
    raw_df["Preferred Entry Date 1"].max(),
)

# Couldn't check min and max for the other two columns because they have NaN values
# print('Preferred Entry Date 2:', raw_df['Preferred Entry Date 2'].min(), 'to', raw_df['Preferred Entry Date 2'].max())
# print('Preferred Entry Date 3:', raw_df['Preferred Entry Date 3'].min(), 'to', raw_df['Preferred Entry Date 3'].max())

Preferred Entry Date 1: 2021-05-15 00:00:00 to 2021-10-30 00:00:00


In [9]:
# Look at values for Results Status column
raw_df["Results Status"].value_counts()

Preferred Entry Date 1   Preferred Division 1  \
136               2021-07-03  Core Enchantment Zone   
947               2021-09-16  Core Enchantment Zone   
1778              2021-07-01  Core Enchantment Zone   
2319              2021-07-01  Core Enchantment Zone   
2342              2021-06-18  Core Enchantment Zone   
...                      ...                    ...   
33685             2021-09-16  Core Enchantment Zone   
35270             2021-09-05  Core Enchantment Zone   
35716             2021-05-18  Core Enchantment Zone   
35862             2021-10-29  Core Enchantment Zone   
36081             2021-08-14  Core Enchantment Zone   

       Minimum Acceptable Group Size 1  Maximum Requested Group Size 1  \
136                                  6                               6   
947                                  2                               2   
1778                                 2                               2   
2319                                 6                               6   
2342                                 6                               6   
...                                ...                             ...   
33685                                4                               4   
35270                                1                               1   
35716                                2                               2   
35862                                6                               6   
36081                                8                               8   

      Preferred Entry Date 2   Preferred Division 2  \
136                 9/4/2021  Core Enchantment Zone   
947                8/19/2021  Core Enchantment Zone   
1778                6/2/2021  Core Enchantment Zone   
2319               8/10/2021  Core Enchantment Zone   
2342               6/25/2021  Core Enchantment Zone   
...                      ...                    ...   
33685              9/23/2021  Core Enchantment Zone   
35270              9/12/2021  Core Enchantment Zone   
35716              7/12/2021  Core Enchantment Zone   
35862              10/8/2021              Snow Zone   
36081              8/21/2021  Core Enchantment Zone   

       Minimum Acceptable Group Size 2  Maximum Requested Group Size 2  \
136                                6.0                             6.0   
947                                4.0                             4.0   
1778                               1.0                             1.0   
2319                               6.0                             6.0   
2342                               5.0                             5.0   
...                                ...                             ...   
33685                              4.0                             4.0   
35270                              1.0                             1.0   
35716                              2.0                             2.0   
35862                              6.0                             6.0   
36081                              8.0                             8.0   

      Preferred Entry Date 3   Preferred Division 3  \
136                10/9/2021  Core Enchantment Zone   
947                5/27/2021  Core Enchantment Zone   
1778               8/25/2021  Core Enchantment Zone   
2319               9/19/2021  Core Enchantment Zone   
2342                6/6/2021  Core Enchantment Zone   
...                      ...                    ...   
33685              9/30/2021  Core Enchantment Zone   
35270              9/19/2021  Core Enchantment Zone   
35716               8/4/2021  Core Enchantment Zone   
35862                    NaN                    NaN   
36081              8/28/2021  Core Enchantment Zone   

       Minimum Acceptable Group Size 3  Maximum Requested Group Size 3  \
136                                6.0                             6.0   
947                                2.0                             2.0   
1778                   

In [5]:
meZone = raw_df["Preferred Division 1"] == "Colchuck Zone"
meDate = raw_df["Preferred Entry Date 1"] == "2021-08-27"
meGroupSize = raw_df["Minimum Acceptable Group Size 1"] == 4

# Search for my application in raw_df
raw_df[meZone & meDate & meGroupSize].head(1)

# That's my application! Loser!

Preferred Entry Date 1 Preferred Division 1  \
931             2021-08-27        Colchuck Zone   

     Minimum Acceptable Group Size 1  Maximum Requested Group Size 1  \
931                                4                               4   

    Preferred Entry Date 2 Preferred Division 2  \
931                    NaN                  NaN   

     Minimum Acceptable Group Size 2  Maximum Requested Group Size 2  \
931                              NaN                             NaN   

    Preferred Entry Date 3 Preferred Division 3  \
931                    NaN                  NaN   

     Minimum Acceptable Group Size 3  Maximum Requested Group Size 3  \
931                              NaN                             NaN   

    Results Status  Awarded Preference Awarded Entry Date  \
931   Unsuccessful                 NaN                NaN   

    Awarded Entrance Code/Name  Awarded Group Size  
931                        NaN                 NaN

In [6]:
meZone1 = raw_df["Preferred Division 1"] == "Colchuck Zone"
meDate1 = raw_df["Preferred Entry Date 1"] == "2021-08-27"

meZone2 = raw_df["Preferred Division 2"] == "Colchuck Zone"
meDate2 = raw_df["Preferred Entry Date 2"] == "2021-08-27"

meZone3 = raw_df["Preferred Division 3"] == "Colchuck Zone"
meDate3 = raw_df["Preferred Entry Date 3"] == "2021-08-27"

# Search for my application in raw_df
(
    raw_df[meZone1 & meDate1].count()
    + raw_df[meZone2 & meDate2].count()
    + raw_df[meZone3 & meDate3].count()
)

Preferred Entry Date 1             53
Preferred Division 1               53
Minimum Acceptable Group Size 1    53
Maximum Requested Group Size 1     53
Preferred Entry Date 2             51
Preferred Division 2               51
Minimum Acceptable Group Size 2    51
Maximum Requested Group Size 2     51
Preferred Entry Date 3             50
Preferred Division 3               50
Minimum Acceptable Group Size 3    50
Maximum Requested Group Size 3     50
Results Status                     53
Awarded Preference                  1
Awarded Entry Date                  1
Awarded Entrance Code/Name          1
Awarded Group Size                  1
dtype: int64

## [Enchantments Permit Lottery Statistics](https://www.fs.usda.gov/Internet/FSE_DOCUMENTS/fseprd568443.pdf)

-   Year - Total Applications Submitted
-   2022 - 36,827 (One application per person)
-   2021 - 39,695 (One application per person)
-   2020 - no data
-   2019 - 24,614 (One application per person)
-   2018 - 21,605 (One application per person)
-   2017 - 20,920 (One application per person)
-   2016 - 19,646 (Up to 3 applications per person)
-   2015 - 12,034 (Up to 3 applications per person)
-   2014 - 8,588 (Up to 3 applications per person)
-   2013 - 4,391 (Up to 3 applications per person)
-   2012 - 2,330 (Unlimited applications per person)

Total Permits Awarded in 2019 - 2,060

Total Permits Awarded in 2021 - 2,444

Total Permits Awarded in 2022 - 2,528


In [7]:
# View percent increase in applications from 2019 to 2021
(39695 / 24614 - 1) * 100

61.27000893800276

## Lottery Information

Applicants were allowed to choose three different entry dates for their trip, group sizes, and permit zone. For example:

-   7/25/2021, Core Enchantment Zone, 4 people
-   8/16/2021, Core Enchantment Zone, 2 people
-   9/2/2021, Stuart Lake Zone, 8 people

However, they didn't have to choose three options. Some applicants only chose to submit one option. Just looking at the `Preferred Entry Date 1`, applicants applied for trips between May 15 and October 30. Applicants chose group sizes between 1-8.

A total of 36,695 applications made up the lottery in 2021 (the website PDF says 39,695 but the data spreadsheet only has 36,695 rows, I believe the website PDF is a typo).

### How Lottery/Quota Works

Per [recreation.gov](https://www.recreation.gov/permits/233273):

> A group may consist of up to eight people, must camp in the Zone indicated on their permit for up to 14 days. The number of overnight groups/people that may enter each day is split by zone during permit season:

-   \*Core Enchantments Zone : 24 people per day (cannot increase group size after lottery)
-   Colchuck Lake Zone: three groups per day
-   Stuart Lake Zone: four groups per day
-   Snow Lakes Zone: five groups per day
-   Eightmile/Caroline Zone: three groups per day

### Awarded Permits

Just because a permit is awarded, doesn't mean that the applicant went on the trip. Looking at the `Results Status` there are possible values of:

-   Unsuccessful
-   Accepted
-   Cancelled
-   No Response
-   Declined

**In 2021, 352 applications didn't response to their awarded lottery, which they had won!** This statistic is even more shocking because I applied for a lottery in 2021 and was `Unsuccessful`. I found my application in row `931` (932 if not using a zero-index). Oddly, there were only 53 other applicants that applied for the same entry date and zone that I did. I'll have to double that after cleaning the data, but that'd be a surprisingly low number for a Friday in late August for the Colchuck Zone.

Equally interesting is that if hundreds go unnclaimed and, as stated on the website, _All unclaimed lottery awards will be returned to the advance reservation system April 1 around 7a.m. PT._ There should be a decent size chunk of permits available for reservation to the fastest and luckiest _clickers_.

Again, I found a very slight difference in "Awarded" permits as the website PDF displayed the statistics. The PDF (linked above) said 2,444 permits were awarded in 2021. Looking at the data in `Results Status`:

-   Unsuccessful 33800
-   Accepted 2036
-   Cancelled 450
-   No Response 352
-   Declined 57

If you add `Accepted` + `No Response` + `Declined` you get 2,445, which is one more than the stated number of 2,444. I'll have to double check the numbers after cleaning, but this is an oddity if not another mistake.

### Lottery Algorithm


## US Forest Service Recommendations

### How can I improve my chances of getting a permit?

> "Because of the lottery, whether or not you are successful in getting a permit is partly luck. Bear in mind that the most popular time to go is August, and the most popular days to start a trip are Fridays, Thursdays, and Saturdays. If you really want to do a Friday-Sunday trip in mid-August, by all means apply for that trip, but remember that you’re odds of getting a permit will be less than if you tried for a Monday-Wednesday trip in July. View lottery statistics here"


## Online Recommendations

### [The Enchantments Lottery: Best Ways to Win a Permit (2023)](https://trekkinghigher.com/enchantments-lottery/)

A lot of just basic information re-iterated from recreation.gov and fs.usda.gov

> Competition for Enchantments permits is heavy. Knowing what permit zones are competitive and when they’re competitive can help improve your permit lottery odds.
>
> In 2022, 36,827 applicants entered the annual permit lottery. The total Enchantment permits awarded were 2,528. The Core Enchantment Zone was far and away the most popular and most competitive.
>
> Snow Zone: 2,881 applicants for 705 permits (24%)
>
> Core Enchantment Zone: 26,988 applicants for 2,920 people (not groups)
>
> Assuming all Core Zone groups were for two people, about 5.4%
>
> Colchuck Zone: 4,606 applicants for 365 permits
>
> Stuart Zone: 1,754 applicants for 535 permits
>
> Eightmile/Caroline Zone: 598 applicants for 365 permits

> **How do you win the lottery for Enchantments?**, The best way to win the Enchantments lottery is to apply for the Colchuck Zone or Snow Zone and avoid stiff competition for the Core Zone. Applying for a Monday, Tuesday, or Wednesday is your best bet at winning the permit lottery for the Core Enchantments Zone. August is the busiest month.

### [Ultimate Enchantments Lottery Guide: What to know for your 2024 trip](https://explorewithalec.com/enchantments-lottery-guide/)

> The first one is choosing dates that are early or late in the backpacking season. Fewer people will apply for these times, increasing your chances of being selected.

> Furthermore, you could choose late into October but might be contending with winter conditions. Please note that the end of September and early October is larch season in Washington, and this is one of the prettiest spots for fall colors.

> As you read above, the core zone Enchantments lottery is based on people, while other areas are based on groups. If you are aiming to land the core permits, picking a smaller group increases your odds. However, it doesn’t matter with other sections as they simply choose a group.

> As you can assume, more people apply for dates overlapping with the weekend to decrease the number of vacation days they have to take. If you happen to land one of these coveted weekend dates, more power to you.
> But if you want to increase your odds, choosing a starting date on a Tuesday or Wednesday may help you out…slightly.

> **Don’t Aim for the Core Zone during the Enchantments Lottery**, Everyone aims for the core zone. So, don’t aim for the core zone region if you want to win. Pick another section and go from there.

### [The Enchantments Lottery Cheat Codes](https://outdoorstatus.com/articles/enchantments-lottery/)

> In 2023, you only had a 6% chance of getting a permit.

-   Eightmile - 12%
-   Stuart - 7%
-   Colchuck - 2%
-   Core - 1% - Smaller is better (max 8 people)
-   Snow - 5%

**By Day**

-   Sun - 4%
-   Mon - 3%
-   Tue - 3%
-   Wed - 3%
-   Thu - 2%
-   Fri - 1%
-   Sat - 2%

**By Month**

-   May - 18%
-   June - 5%
-   Jul - 2%
-   Aug - 1%
-   Sep - 2%
-   Oct - 7%


## What questions or claims do we want to answer or test?

-   Does applying for a zone other than core increase your chance of winning the lottery?
-   Does applying other than Thurs, Fri, Sat increase your chance of winning the lottery?
-   Which preference was awarded the most? Or how many people were awarded a core enchantments on their first option?
-   What's your chance of winning the lottery?
-   How much does applying for a zone ther than Core, during non Thurs, Fri, Sat days increase your chances?
-   Can we verify that lottery is working how it says?
-   Did group size in Core effect awarded permits?
